extensions/wikia/Track/Track.php

```diff
@@ -1,59 +1,131 @@
 
-$wgExtensionCredits['other'][] = array(
+$wgExtensionCredits[ 'other' ][] = array(
 	'name' => 'Track',
 	'author' => 'Garth Webb',
 	'description' => 'A class to help track events and pageviews for wikia'
 );
 
-$wgHooks['MakeGlobalVariablesScript'][] = 'Track::addGlobalVars';
-$wgHooks['WikiaSkinTopScripts'][] = 'Track::onWikiaSkinTopScripts';
+$wgHooks[ 'MakeGlobalVariablesScript' ][] = 'Track::addGlobalVars';
+$wgHooks[ 'WikiaSkinTopScripts' ][] = 'Track::onWikiaSkinTopScripts';
 
 class Track {
 	const BASE_URL = 'http://a.wikia-beacon.com/__track';
+	const GA_URL = 'https://www.google-analytics.com';
+	const GA_VERSION = 1;
 
-	private static function getURL ($type=null, $name=null, $param=null, $for_html=true) {
+	private static function getURL( $type = null, $name = null, $param = null, $for_html = true ) {
 		global $wgStyleVersion, $wgCityId, $wgContLanguageCode, $wgDBname, $wgDBcluster, $wgUser, $wgArticle, $wgTitle, $wgAdServerTest;
 
 		$sep = $for_html ? '&amp;' : '&';
 
-		$url = Track::BASE_URL.
-			($type ? "/$type" : '').
-			($name ? "/$name" : '').
-			'?'.
-			'cb='.$wgStyleVersion.$sep.
-			'c='.$wgCityId.$sep.
-			'lc='.$wgContLanguageCode.$sep.
-			'lid='.WikiFactory::LangCodeToId($wgContLanguageCode).$sep.
-			'x='.$wgDBname.$sep.
-			'y='.$wgDBcluster.$sep.
-			'u='.$wgUser->getID().$sep.
-			'a='.(is_object($wgArticle) ? $wgArticle->getID() : null).$sep.
-			's='.RequestContext::getMain()->getSkin()->getSkinName().
-			($wgTitle && !is_object($wgArticle) ? $sep.'pg='.urlencode($wgTitle->getPrefixedDBkey()) : '').
-			($wgTitle ? $sep.'n='.$wgTitle->getNamespace() : '').
-			(!empty($wgAdServerTest) ? $sep.'db_test=1' : '');
+		$url = Track::BASE_URL .
+			   ( $type ? "/$type" : '' ) .
+			   ( $name ? "/$name" : '' ) .
+			   '?' .
+			   'cb=' . $wgStyleVersion . $sep .
+			   'c=' . $wgCityId . $sep .
+			   'lc=' . $wgContLanguageCode . $sep .
+			   'lid=' . WikiFactory::LangCodeToId( $wgContLanguageCode ) . $sep .
+			   'x=' . $wgDBname . $sep .
+			   'y=' . $wgDBcluster . $sep .
+			   'u=' . $wgUser->getID() . $sep .
+			   'a=' . ( is_object( $wgArticle ) ? $wgArticle->getID() : null ) . $sep .
+			   's=' . RequestContext::getMain()->getSkin()->getSkinName() .
+			   ( $wgTitle && !is_object( $wgArticle ) ? $sep . 'pg=' . urlencode( $wgTitle->getPrefixedDBkey() ) : '' ) .
+			   ( $wgTitle ? $sep . 'n=' . $wgTitle->getNamespace() : '' ) .
+			   ( !empty( $wgAdServerTest ) ? $sep . 'db_test=1' : '' );
 
 		// Handle any parameters passed to us
-		if ($param) {
-			foreach ($param as $key => $val) {
-				$url .= $sep.urlencode($key).'='.urlencode($val);
+		if ( $param ) {
+			foreach ( $param as $key => $val ) {
+				$url .= $sep . urlencode( $key ) . '=' . urlencode( $val );
 			}
 		}
 
 		return $url;
 	}
 
-	private static function getViewJS ($param=null) {
+	private static function getGAURL( $type, $category, $action, $label, $value, $tid, $extraParams = [ ], $for_html = null ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This function has 8 parameters, which is greater than the 7 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS107)